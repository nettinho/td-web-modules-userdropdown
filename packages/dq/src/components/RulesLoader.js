import React from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Loading } from "@truedat/core/components";
import { clearRules, fetchRules } from "../routines";

export class RulesLoader extends React.Component {
  static propTypes = {
    fetchRules: PropTypes.func,
    clearRules: PropTypes.func,
    rulesLoading: PropTypes.bool
  };

  componentDidMount() {
    const { fetchRules } = this.props;
    fetchRules();
  }

  componentWillUnmount() {
    const { clearRules } = this.props;
    clearRules();
  }

  render() {
    const { rulesLoading } = this.props;

    if (rulesLoading) {
      return <Loading />;
    } else {
      return null;
    }
  }
}

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ clearRules, fetchRules }, dispatch)
});

const mapStateToProps = ({ rulesLoading }) => ({ rulesLoading });

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RulesLoader);
