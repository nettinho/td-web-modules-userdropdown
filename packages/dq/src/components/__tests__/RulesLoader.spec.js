import React from "react";
import { shallow } from "enzyme";
import { RulesLoader } from "../RulesLoader";

describe("<RulesLoader />", () => {
  it("matches the latest snapshot", () => {
    const fetchRules = jest.fn();
    const rulesLoading = true;
    const props = { fetchRules, rulesLoading };
    const wrapper = shallow(<RulesLoader {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a loader if rulesLoading is true", () => {
    const fetchRules = jest.fn();
    const rulesLoading = true;
    const props = { fetchRules, rulesLoading };
    const wrapper = shallow(<RulesLoader {...props} />);
    expect(wrapper.find("Loading").length).toBe(1);
  });

  it("renders null if rulesLoading is false", () => {
    const fetchRules = jest.fn();
    const rulesLoading = false;
    const props = { fetchRules, rulesLoading };
    const wrapper = shallow(<RulesLoader {...props} />);
    expect(wrapper.getElement()).toBeNull();
  });

  it("calls fetchRules when component mounts, clearRules when component unmounts", () => {
    const fetchRules = jest.fn();
    const clearRules = jest.fn();
    const rulesLoading = false;
    const props = { fetchRules, rulesLoading, clearRules };
    jest.spyOn(RulesLoader.prototype, "componentDidMount");
    const wrapper = shallow(<RulesLoader {...props} />);
    expect(RulesLoader.prototype.componentDidMount.mock.calls.length).toBe(1);
    expect(clearRules.mock.calls.length).toBe(0);
    expect(fetchRules.mock.calls.length).toBe(1);
    wrapper.unmount();
    expect(clearRules.mock.calls.length).toBe(1);
    expect(fetchRules.mock.calls.length).toBe(1);
  });
});
