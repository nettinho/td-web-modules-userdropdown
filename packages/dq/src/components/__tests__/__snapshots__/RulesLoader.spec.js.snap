// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RulesLoader /> matches the latest snapshot 1`] = `<Loading />`;
